crates/egui-wgpu/src/winit.rs

```diff
@@ -487,7 +491,12 @@ impl Painter {
 
         {
             let renderer = render_state.renderer.read();
-            let frame_view = if capture {
+            let frame_view = if capture
+                & output_frame
+                    .texture
+                    .usage()
+                    .contains(wgpu::TextureUsages::COPY_DST)
```

Comment:
the repeated check on the texture usage is a bit tedious and arcane if one doesn't know about this. How about putting a `supports_screenshots` bool on `surface_state` which we then use to set `capture` always to false + log a warning (I'm on the fence if that's worth a `log::error`, in many cases this isn't critical enough)

Reply:
Great call! On the error / warning debate, I originally wrote this for my application in which I want to be able to export the shown video as an mp4. In this case if the user requests the video to be recorded but it can't be, I would consider it an error - just my 2 cents.